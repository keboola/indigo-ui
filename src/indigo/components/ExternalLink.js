import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const ExternalLink = ({ href, children, className, ...props }) => (
  <a
    {...props}
    href={href}
    target="_blank"
    rel="noopener noreferrer"
    className={classnames(className)}
  >
    {children}
  </a>
);

ExternalLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default ExternalLink;
