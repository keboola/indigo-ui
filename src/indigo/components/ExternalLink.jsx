import React from 'react';
import PropTypes from 'prop-types';

const ExternalLink = ({ href, children, className }) => {
  return (
    <a href={href} target="_blank" rel="noopener noreferrer" className={className}>
      {children}
    </a>
  );
};
ExternalLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

export default ExternalLink;
