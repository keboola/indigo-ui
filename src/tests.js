import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
