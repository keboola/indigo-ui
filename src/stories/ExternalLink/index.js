import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import ExternalLink from '../../indigo/components/ExternalLink';

storiesOf('ExternalLink', module).add(
  'ExternalLink',
  withInfo({
    text: `
        Renders anchor element with additional *target="_blank"* and *rel="noopener noreferrer"* attributes.
        `,
    inline: true,
  })(() => {
    return <ExternalLink href="https://www.keboola.com">keboola.com</ExternalLink>;
  })
);
