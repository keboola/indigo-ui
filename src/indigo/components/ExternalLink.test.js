import React from 'react';

import { snapshot } from '../../tests';
import ExternalLink from './ExternalLink';

describe('<ExternalLink />', () => {
  it('ExternalLink - default', () => {
    snapshot(<ExternalLink href="https://www.keboola.com">keboola.com</ExternalLink>);
  });

  it('ExternalLink - with classname', () => {
    snapshot(
      <ExternalLink href="https://www.keboola.com" className="btn btn-link">
        keboola.com
      </ExternalLink>
    );
  });

  it('ExternalLink - with random props', () => {
    snapshot(
      <ExternalLink
        href="https://www.keboola.com"
        className="btn btn-link"
        id="external-link-id-prop"
      >
        keboola.com
      </ExternalLink>
    );
  });
});
