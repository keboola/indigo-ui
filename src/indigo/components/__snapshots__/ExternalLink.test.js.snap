// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink /> ExternalLink - default 1`] = `
<a
  className=""
  href="https://www.keboola.com"
  rel="noopener noreferrer"
  target="_blank"
>
  keboola.com
</a>
`;

exports[`<ExternalLink /> ExternalLink - with classname 1`] = `
<a
  className="btn btn-link"
  href="https://www.keboola.com"
  rel="noopener noreferrer"
  target="_blank"
>
  keboola.com
</a>
`;

exports[`<ExternalLink /> ExternalLink - with random props 1`] = `
<a
  className="btn btn-link"
  href="https://www.keboola.com"
  id="external-link-id-prop"
  rel="noopener noreferrer"
  target="_blank"
>
  keboola.com
</a>
`;
